lib/samson/build_finder.rb

```diff
@@ -90,7 +90,12 @@ def possible_builds
         commits.unshift previous if previous
       end
 
-      Build.where(git_sha: commits).sort_by { |build| commits.index(build.git_sha) }
+      Build.where(git_sha: commits).
+        group_by { |build| commits.index(build.git_sha) }.
+        sort_by { |index, _| index }.to_h.
+        transform_values { |builds| builds.sort_by!(&:updated_at).reverse }.
+        values.
+        flatten
```

Comment:
can sort by index+updated_at and then uniq by sha ?

Reply:
I struggled with working out the right syntax for that, could you write a diff for that?

Reply:
`Build.where(git_sha: commits).sort_by { |build| [commits.index(build.git_sha), -build.updated_at.to_i]  }`